### PR TITLE
Don't swallow `<> </>` fragments when formatting

### LINKED ISSUE #870: Formatting makes `<></>` fragment dissapear
```
<div onClick=this##handleClick>
  <>foo</>
</div>
```

Turns into

```
<div onClick=this##handleClick>
  foo
</div>

```

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes #870

### CODE REVIEW

File: src/reason-parser/reason_pprint_ast.ml [hunk 1]
```diff
@@ -5629,8 +5629,13 @@ let printer = object(self:'self)
     | {pexp_desc = Pexp_constant constant} as x :: remaining ->
       let raw_literal, _ = extract_raw_literal x.pexp_attributes in
       self#formatChildren remaining (self#constant ?raw_literal constant :: processedRev)
-    | {pexp_desc = Pexp_construct ({txt = Lident "::"}, Some {pexp_desc = Pexp_tuple children} )} :: remaining ->
-      self#formatChildren (remaining @ children) processedRev
+    | {pexp_desc = Pexp_construct ({txt = Lident "::"}, Some {pexp_desc = Pexp_tuple children} )} as x :: remaining ->
+      let {jsxAttrs} = partitionAttributes x.pexp_attributes in
+      if jsxAttrs != [] then
```
Comment (IwanKaramazow) [inline src/reason-parser/reason_pprint_ast.ml]: Another option is the pattern match on `jsxAttrs`.  Pattern matching is probably cheaper than physical equality checks. (Correct me if I'm wrong).

**Comment by IwanKaramazow** [comment]
Can you add some extra tests to verify other expressions?
For example function calls need to be wrapped in braces/parens.

File: src/reason-parser/reason_pprint_ast.ml (hunk 1 above)
Comment (hcarty) [reply src/reason-parser/reason_pprint_ast.ml]: Pattern matching should be safer to against future changes in compiler optimizations around equality.

Comment (anmonteiro) [reply src/reason-parser/reason_pprint_ast.ml]: Hah, I actually had a pattern match at first but decided to go with this approach given it’s used everywhere else that does the same thing.

**Comment by anmonteiro** [comment]
Addressed the comments in the new commits.

**Comment by chenglou** [comment]
Thanks. Will be useful soon I think

**Pull request merged**